docs/content/Caching/Getting-Started-Pre-Aggregations.md

```diff
@@ -159,7 +159,175 @@ the pre-aggregations are rebuilt.
+      format: `currency`,
+    },
+  },
+  ...,
+});
+```
+
+Because neither `quantity` and `price` reference any other measures in our
+`LineItems` cube, we can also say that they are "additive leaf measures". Any
+query requesting only these two measures can be called a "leaf measure additive"
+query.
+
+### Non-Additivity
+
+Using the same sample data for `line_items`, there's a `profit_margin_pct` field
+which is different for each row. However, despite the value being numerical, it
+doesn't actually make sense to add up this value - doing so with the sample data
+would result in a 215% profit margin which (unfortunately) is not correct. We
+can add the `profit_margin_pct` field to our schema like this:
+
+```javascript
+cube(`LineItems`, {
+  ...,
+  measures: {
+    ...,
+    profitMargin: {
+      sql: `profit_margin`,
+      type: `number`,
+      format: 'percentage',
+    },
+  },
+  ...,
+});
+```
+
+Another example of a non-additive measure would be a distinct count of
+`product_id`. If we took the distinct count of products sold over a month, and
+then tried to sum the distinct count of products for each individual day and
+compared them, we would not get the same results. We can add the measure like
+this:
+
+```javascript
+cube(`LineItems`, {
+  ...,
+  measures: {
+    ...,
+    countDistinctProducts: {
+      sql: `product_id`,
+      type: `countDistinct`,
+    },
+  },
+  ...,
+});
+```
+
+### How additivity plays a role in pre-aggregation selection
+
+To recap what we've learnt so far:
+
+- "Additive measures" are measures whose values can be added together.
+- "Multiplied" measures are measures that define `hasMany` relations.
+- "Leaf measures" are measures that do not reference any other measures in their
+  definition.
+- A query is "leaf measure additive" if all of its leaf measures are one of:
+  `count`, `sum`, `min`, `max` or `countDistinctApprox`.
+
+Cube.js selects the best available pre-aggregation based on the incoming queries
+it receives via the API.
+
+Some considerations:
+
+- The query's time dimension and granularity together act as a dimension. If the
+  date range isn't aligned with granularity, a common granularity is used. To
+  match granularity date range, its start date and end date should match. For
+  example, for month it's ['2020-01-01T00:00:00.000', '2020-01-31T23:59:59.999']
+  and for day it's ['2020-01-01T00:00:00.000', '2020-01-01T23:59:59.999']. Date
+  ranges are inclusive. The minimum granularity is `second`.
+- The order in which pre-aggregations are defined in schemas matter; the first
+  matching pre-aggregation for a query is the one that is used. Cubes of a
+  measures and then cubes of dimensions are checked to find a matching `rollup`.
+  However `rollup` pre-aggregations always have priority over `originalSql`.
+  Thus, if you have both `originalSql` and `rollup` defined, Cube.js will try to
+  find matching `rollup` before trying to find matching `originalSql`.
```

Comment:
I wonder if we need to tell anything about `originalSql` pre-aggs here. First, I believe, they should go away. Second, they never are the source of confusion, right? Getting rid of them would help make the explanation shorter and more comprehensible, right?